src/reporters/console/console-reporter.js

```diff
@@ -27,9 +27,10 @@ type Row = Array<string>;
 export default class ConsoleReporter extends BaseReporter {
   constructor(opts: Object) {
     super(opts);
-    this._lastCategorySize = 0;
 
+    this._lastCategorySize = 0;
     this.format = (chalk: any);
+    this.isSilent = !!opts.isSilent;
```

Comment:
I think your solution makes total sense even for install and check commands.
Just add it to your test plan and check what happens if you run other commands with this flag.